core/src/main/scala/za/co/absa/spline/harvester/dispatcher/modelmapper/UntypedMapConverter.scala

```diff
@@ -16,23 +16,19 @@
 
 package za.co.absa.spline.harvester.dispatcher.modelmapper
 
-import scala.language.implicitConversions
+import za.co.absa.spline.producer.model.AttrOrExprRef
 
-object ImplicitMappingConversions {
-  implicit def stringToOption(str: String): Option[String] = Option(str)
+class UntypedMapConverter(toAttrOrExprRef: (AttrOrExprRef => Any)) {
 
-  implicit def seqToOption[T](seq: Seq[T]): Option[Seq[T]] = {
-    if (seq.isEmpty) {
-      None
-    } else {
-      Some(seq)
+  def toUntypedMap(map: Map[String, Any]): Map[String, Any] = {
+    def mapValue(value: Any): Any = value match {
+      case aoe: AttrOrExprRef => toAttrOrExprRef(aoe)
+      case s: Seq[_] => s.map(mapValue)
+      case m: Map[_, _] => m.transform((k, v) => mapValue(v))
+      case o: Option[_] => o.map(mapValue)
+      case other => other
     }
-  }
 
-  implicit def mapToOption[K, V](map: Map[K, V]): Option[Map[K, V]] =
-    if (map.isEmpty) {
-      None
-    } else {
-      Some(map)
-    }
+    map.transform((k, v) => mapValue(v))
```

Comment:
`k` is unused. Wouldn't  `map.mapValue()` be enough? (same above)

Reply:
I don't like how it hides exceptions https://dev.to/cerveada/scala-tip-mapvalues-may-not-do-what-you-expect-2n2e

Reply:
hide? Do you mean throw it on a later stage?

Reply:
yes

Reply:
Got it. Just a minor suggestion - replace unused arguments with `_` for easier reading.